Lombiq.Tests.UI/Extensions/OrchardCoreDashboardUITestContextExtensions.cs

```diff
@@ -15,26 +16,47 @@ public static void ClickPublish(this UITestContext context, bool withJavaScript
         {
             if (withJavaScript)
             {
-                context.ExecuteScript($"document.querySelector('.publish-button').click()");
+                context.ExecuteScript("document.querySelector('.publish-button, .publish.btn').click();");
             }
             else
             {
                 context.ClickReliablyOn(By.Name("submit.Publish"));
             }
         }
 
-        public static void GoToContentItemList(this UITestContext context)
+        public static void ClickModalOk(this UITestContext context) => context.ClickReliablyOn(By.Id("modalOkButton"));
+
+        /// <summary>
+        /// Sometimes the Publish button doesn't get clicked. This method retries pressing it up to 4 times with a 30
+        /// second interval between attempts. This should grant enough time to execute the publish action if the button
+        /// actually got pressed.
+        /// </summary>
+        public static void ClickPublishReliably(this UITestContext context, bool withJavaScript = false)
         {
-            context.ClickReliablyOn(By.CssSelector("#content"));
-            context.ClickReliablyOn(By.LinkText("Content Items"));
+            var navigationState = context.AsPageNavigationState();
+
+            context.DoWithRetriesOrFail(
+                () =>
+                {
+                    ClickPublish(context, withJavaScript);
+                    return navigationState.CheckIfNavigationHasOccurred();
+                },
+                timeout: TimeSpan.FromSeconds(30),
+                interval: TimeSpan.FromMinutes(2));
```

Comment:
These should rather default to the values coming from the config (like `DoWithRetriesOrFail()` does) or at least be possible to override.